contracts/interchain-token-service/src/contract/execute/mod.rs

```diff
@@ -182,13 +181,17 @@ fn apply_to_hub(
     .then(Result::Ok)
 }
 
-fn apply_to_register_token(
+fn execute_register_token(
     storage: &mut dyn Storage,
     source_chain: ChainNameRaw,
     register_token: RegisterTokenMetadata,
-) -> Result<RegisterTokenMetadata, Error> {
+) -> Result<Response, Error> {
+    ensure_chain_not_frozen(storage, &source_chain)?;
     interceptors::register_custom_token(storage, source_chain, register_token.clone())?;
-    Ok(register_token)
+    Ok(Response::new().add_event(Event::TokenMetadataRegistered {
```

Comment:
```suggestion
    ensure_chain_not_frozen(storage, &source_chain)?;

    interceptors::register_custom_token(storage, source_chain, register_token.clone())?;

    Ok(Response::new().add_event(Event::TokenMetadataRegistered {
```
almost missed the frozen check haha